geonode/maps/qgis_server_views.py

```diff
@@ -632,6 +634,18 @@ def set_thumbnail_map(request, mapid):
 
     # Give thumbnail creation to celery tasks, and exit.
     map_obj = Map.objects.get(id=mapid)
+    # Re-create the qgs project
+    qgis_map, created = QGISServerMap.objects.get_or_create(map=map_obj)
+    response = create_qgis_project(
```

Comment:
Why we need to execute this? If the map object already exists, then QGIS Project should be already exists. If user add/change new layers, then it will need to be reexecute it on post_save signals for Map.

Reply:
Yes, we need to regenerate the QGIS project, especially if layers are changed in the map. But I agree with whit Rizky, why not in the signal?